src/OrleansAWSUtils/Membership/SiloInstanceRecord.cs

```diff
@@ -0,0 +1,210 @@
+
+            if (fields.ContainsKey(SILO_IDENTITY_PROPERTY_NAME))
+                SiloIdentity = fields[SILO_IDENTITY_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(ADDRESS_PROPERTY_NAME))
+                Address = fields[ADDRESS_PROPERTY_NAME].S;
+
+            int port;
+            if (fields.ContainsKey(PORT_PROPERTY_NAME) && 
+                int.TryParse(fields[PORT_PROPERTY_NAME].N, out port))
+                Port = port;
+
+            int generation;
+            if (fields.ContainsKey(GENERATION_PROPERTY_NAME) &&
+                int.TryParse(fields[GENERATION_PROPERTY_NAME].N, out generation))
+                Generation = generation;
+
+            if (fields.ContainsKey(HOSTNAME_PROPERTY_NAME))
+                HostName = fields[HOSTNAME_PROPERTY_NAME].S;
+
+            int status;
+            if (fields.ContainsKey(STATUS_PROPERTY_NAME) &&
+                int.TryParse(fields[STATUS_PROPERTY_NAME].N, out status))
+                Status = status;
+
+            int proxyPort;
+            if (fields.ContainsKey(PROXY_PORT_PROPERTY_NAME) &&
+                int.TryParse(fields[PROXY_PORT_PROPERTY_NAME].N, out proxyPort))
+                ProxyPort = proxyPort;
+
+            if (fields.ContainsKey(SILO_NAME_PROPERTY_NAME))
+                SiloName = fields[SILO_NAME_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(SUSPECTING_SILOS_PROPERTY_NAME))
+                SuspectingSilos = fields[SUSPECTING_SILOS_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(SUSPECTING_TIMES_PROPERTY_NAME))
+                SuspectingTimes = fields[SUSPECTING_TIMES_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(START_TIME_PROPERTY_NAME))
+                StartTime = fields[START_TIME_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(I_AM_ALIVE_TIME_PROPERTY_NAME))
+                IAmAliveTime = fields[I_AM_ALIVE_TIME_PROPERTY_NAME].S;
+
+            int etag;
+            if (fields.ContainsKey(ETAG_PROPERTY_NAME) &&
+                int.TryParse(fields[ETAG_PROPERTY_NAME].N, out etag))
+                ETag = etag;
+        }
+        
+        public string DeploymentId { get; set; }
+        public string SiloIdentity { get; set; }
+        public string Address { get; set; }
+        public int Port { get; set; }
+        public int Generation { get; set; }
+        public string HostName { get; set; }
+        public int Status { get; set; }
+        public int ProxyPort { get; set; }
+        public string SiloName { get; set; }
+        public string SuspectingSilos { get; set; }
+        public string SuspectingTimes { get; set; }
+        public string StartTime { get; set; }
+        public string IAmAliveTime { get; set; }
+        public int ETag { get; set; }        
+
+        internal static SiloAddress UnpackRowKey(string rowKey)
+        {
+            var debugInfo = "UnpackRowKey";
+            try
+            {
+#if DEBUG
```

Comment:
Could these be logged on `Debug` level. Maybe the idea here is that this class doesn't take logger as a parameter, but maybe the code using this could log relevant pieces.